src/cmd/linuxkit/run_qemu.go

```diff
@@ -525,13 +526,19 @@ func buildQemuCmdline(config QemuConfig) (QemuConfig, []string) {
 	if config.NetdevConfig == "" {
 		qemuArgs = append(qemuArgs, "-net", "none")
 	} else {
-		mac := generateMAC()
-		qemuArgs = append(qemuArgs, "-net", "nic,model=virtio,macaddr="+mac.String())
+		var macRemoveRegExp = regexp.MustCompile(",((?:[0-9A-Fa-f]{2}[:-]){5}[0-9A-Fa-f]{2})|((?:[0-9A-Fa-f]{2}[:-]){5}[0-9A-Fa-f]{2}),")
+		var macRegExp = regexp.MustCompile("([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})")
+		var mac = macRegExp.FindString(config.NetdevConfig)
```

Comment:
I did not know this function but i dont think its very helpful. It seems to take a string which must be a valid mac and returns a type HardwareAddr. In the actual case we get something like -networking tap,lin0,22:11:11:11:11:11 and need to parse it to the variables mac=22:11:11:11:11:11 and netdevConfig=tap,lin0. the remove Regex also correctly parses tap,22:11:11:11:11:11,lin0 or 22:11:11:11:11:11,lin0,tap format by Regex duplication